diesel-wasm-sqlite/src/connection/raw.rs

```diff
@@ -85,37 +85,49 @@ impl RawConnection {
         flags as i32
     }
 
-    /* possible to implement this, but would need to fill in the missing wa-sqlite functions
-    pub(super) fn serialize(&mut self) -> SerializedDatabase {
-        unsafe {
-            let mut size: ffi::sqlite3_int64 = 0;
-            let data_ptr = ffi::sqlite3_serialize(
-                self.internal_connection.as_ptr(),
-                std::ptr::null(),
-                &mut size as *mut _,
-                0,
-            );
-            SerializedDatabase::new(data_ptr, size as usize)
+    pub(super) fn serialize(&self, schema: &str, flags: u32) -> SerializedDatabase {
+        let sqlite3 = crate::get_sqlite_unchecked();
+        let mut p_size: i64 = 0;
```

Comment:
You're going to have to pass a pstack pointer here, so that SQLite can write `serialize` to it within it's WASM memory module


In this project, pstack pointers are only used in `Statement::prepare`. Documentation for them is also [here](https://sqlite.org/wasm/doc/trunk/api-wasm.md#wasm-pstack)

Once you get the size, you should be able to use `ffi::raw_copy_from_sqlite` to copy from the sqlite memory back into our host memory.